tools/python/boutiques/searcher.py

```diff
@@ -0,0 +1,46 @@
+import requests
+from tabulate import tabulate
+
+
+class ZenodoError(Exception):
+    pass
+
+
+class Searcher():
+
+    def __init__(self, query, verbose):
+        if query is not None:
+            self.query = query
+        else:
+            self.query = 'boutiques'
+
+        self.verbose = verbose
+
+    def search(self):
+        r = requests.get('https://zenodo.org/api/records/?q=%s&'
+                         'keywords=boutiques&keywords=schema&'
+                         'keywords=version&file_type=json&type=software'
+                         % self.query)
+        if(r.status_code != 200):
+            self.raise_zenodo_error("Error searching Zenodo", r)
+
+        if(self.verbose):
+            self.print_zenodo_info("Zenodo search returned %d results"
+                                   % r.json()["hits"]["total"], r)
+
+        return self.create_results_table(r.json())
+
+    def create_results_table(self, results):
+        table = []
+        for hit in results["hits"]["hits"]:
+            table.append([hit["id"], hit["metadata"]["title"],
+                         hit["metadata"]["description"]])
+        return tabulate(table, headers=['ID', 'TITLE',
+                        'DESCRIPTION'], tablefmt='plain')
```

Comment:
A verbose mode could return more info, such as:
* Tags
* Author
* Container ('Docker', 'Singularity' or 'None')
* Schema version
* DOI
* Tests y/n
* Tool version
* Descriptor version
Some of these things should be accessible through Zenodo without having to download the descriptor. For the other ones, we could add explicit Zenodo tags in the publisher (create an issue).

Reply:
Also, verbose could list all the versions of the tools, while non-verbose could aggregate them, as it is the case now.

Reply:
Actually non-verbose should report only the latest versions.

Reply:
Aren't tool version and descriptor version the same thing (at least with the way publisher is currently implemented)?

Reply:
Also, getting all versions of the tool isn't really possible through the search API. The JSON returned by the search only lists the current version's ID, the number of versions and the original version's ID. Should I include any of those?

Reply:
The tabular format looks a bit messy on my little VM with all those columns:
![image](https://user-images.githubusercontent.com/14361988/47932100-1d765880-dea7-11e8-8c96-b39ca6e5b386.png)
Should I keep it this way or change the format?

Reply:
> Aren't tool version and descriptor version the same thing (at least with the way publisher is currently implemented)?

No they are different. The tool is what is described by the descriptor.

Reply:
> Also, getting all versions of the tool isn't really possible through the search API. The JSON returned by the search only lists the current version's ID, the number of versions and the original version's ID. Should I include any of those (number of versions or parent ID)?

so is there any way to find the zenodo id of a previous version if we wanted to get back to it? I assume this might require a bit more work, perhaps add an issue on this so that we can get the current implementation, which is already awesome, merged rapidly.

Reply:
> The tabular format looks a bit messy on my little VM with all those columns:
> ![image](https://user-images.githubusercontent.com/14361988/47932100-1d765880-dea7-11e8-8c96-b39ca6e5b386.png)
> Should I keep it this way or change the format?

Perhaps you could truncate the description and keywords to a max number of chars but I don't find it a very big deal.

Reply:
> > Aren't tool version and descriptor version the same thing (at least with the way publisher is currently implemented)?
> 
> No they are different. The tool is what is described by the descriptor.

That makes sense but currently the publisher sets the metadata as `'version': self.descriptor['tool-version']`. Should this be changed?

Reply:
The most important version is the tool version indeed, so that should be the version in Zenodo. Now, what happens if one publishes multiple descriptors with the same tool version and name? How do they differ in Zenodo?

Reply:
Only the ID will differ. Zenodo allows for multiple records with the same name and version. An example from Sandbox:
![image](https://user-images.githubusercontent.com/14361988/47941855-36dacd00-dec6-11e8-81c4-2dfc2fb90992.png)

Reply:
The search API only returns the latest version of each record, though.

Reply:
ok then so be it!